otx/algorithms/classification/tasks/inference.py

```diff
@@ -283,25 +283,41 @@ def _init_recipe(self):
         train_type = self._hyperparams.algo_backend.train_type
         logger.info(f"train type = {train_type}")
 
-        if train_type == TrainType.INCREMENTAL:
-            recipe = os.path.join(recipe_root, "class_incr.yaml")
-        if train_type == TrainType.INCREMENTAL and self._multilabel:
+        # load INCREMENTAL recipe file first. (default train type)
+        if self._multilabel:
             recipe = os.path.join(recipe_root, "class_incr_multilabel.yaml")
         else:
-            if train_type == TrainType.SEMISUPERVISED:
-                raise NotImplementedError(f"train type {train_type} is not implemented yet.")
-            if train_type == TrainType.SELFSUPERVISED:
-                raise NotImplementedError(f"train type {train_type} is not implemented yet.")
-            # raise NotImplementedError(f'train type {train_type} is not implemented yet.')
-            # FIXME: Temporary remedy for CVS-88098
             recipe = os.path.join(recipe_root, "class_incr.yaml")
-            logger.warning(f"train type {train_type} is not implemented yet. Running incremental training.")
+
+        if train_type != TrainType.INCREMENTAL:
```

Comment:
The order of these if statements doesn't seem appropriate. My suggestion is to check the semi first, and then the incremental check separately. 
And in the case of multi_label (or h-label), What recipe does semi-Multilabel use? if it is not currently supported, we need to raise NotImplemented or change to INCREMENTAL